okio/src/test/java/okio/PipeTest.java

```diff
@@ -357,6 +372,195 @@
     assertElapsed(4000.0, start);
   }
 
+  @Test public void fold() throws IOException {
+    final Pipe pipe = new Pipe(128);
+
+    final BufferedSink pipeSink = Okio.buffer(pipe.sink());
+    pipeSink.writeUtf8("hello");
+    pipeSink.emit();
+
+    final BufferedSource pipeSource = Okio.buffer(pipe.source());
+    assertEquals("hello", pipeSource.readUtf8(5));
+
+    final Buffer foldedSinkBuffer = new Buffer();
+    final boolean[] foldedSinkClosed = {false};
```

Comment:
For what it’s worth, AtomicBoolean is much nicer for this job.

Reply:
Thank you! The IDE suggested I turned to bool into an array and I didn't think twice